platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultResolvedDependency.java

```diff
@@ -129,9 +136,9 @@ public Set<ResolvedArtifact> getParentArtifacts(ResolvedDependency parent) {
 
     private Set<ResolvedArtifact> sort(ResolvedArtifactSet artifacts) {
         ArtifactCollectingVisitor visitor = new ArtifactCollectingVisitor(new TreeSet<>(new ResolvedArtifactComparator()));
-        ParallelResolveArtifactSet.wrap(artifacts, buildOperationProcessor).visit(visitor);
+        ParallelResolveArtifactSet.wrap(artifacts, buildOperationExecutor).visit(visitor);
         if (!visitor.getFailures().isEmpty()) {
-            throw UncheckedException.throwAsUncheckedException(visitor.getFailures().get(0));
+            resolutionHost.rethrowFailure("artifacts", visitor.getFailures());
         }
         return visitor.getArtifacts();
     }
```

Comment:
🤔 This method is bad.  

- It's a `sort` that returns a `Set`, called by methods that return `Set`s themselves.  There's nothing in the public API or documentation that promises to sort these.  It should at least be documented until we can return a `SortedSet` with Java 11.

- It is weird that sorting something has the potential of throwing failures.  Sorting intuitively just shouldn't do that.

- It's also weird that `ResolvedArtifactSet` doesn't offer an `asSortedSet` method directly.

- It's also very weird that this method is ultimately what causes the artifacts to be resolved...and it's named `sort` with no explanation of this.  Sorting is like the least important thing it's doing.  It should at least be renamed - separating the sorting and resolving functions completely would be better, but that looks tougher.

Reply:
Like I said above. This is all legacy code. I don't think we should put any effort into making it better if we intend to remove it

Reply:
We can rename the method if that makes you happier but I do not think we should spend much time here

Reply:
This is fine.